web_src/js/components/RepoContributors.vue

```diff
@@ -0,0 +1,249 @@
+      :data="toGraphData(totalStats.weeks, 'main')"
+      :options="getOptions('main')"
+    />
+
+    <div class="ui attached segment two column grid">
+      <div
+        v-for="(contributor, index) in sortedContributors"
+        :key="index"
+        class="column stats-table"
+      >
+        <div class="ui top attached header gt-df gt-f1">
+          <b class="ui right">#{{ index + 1 }}</b>
+          <a :href="contributor.home_link">
+            <img
+              height="40"
+              width="40"
+              :href="contributor.avatar_link"
+              :src="contributor.avatar_link"
+            >
+          </a>
+          <div class="gt-ml-3">
+            <a :href="contributor.home_link"><h4>{{ contributor.name }}</h4></a>
+            <p class="gt-font-12">
+              <strong>{{ contributor.total_commits.toLocaleString() }} commits
+              </strong>
+              <strong class="text green">{{ additions(contributor.weeks).toLocaleString() }}++
+              </strong>
+              <strong class="text red">
+                {{ deletions(contributor.weeks).toLocaleString() }}--</strong>
+            </p>
+          </div>
+        </div>
+        <div class="ui attached segment">
+          <Line
+            :data="toGraphData(contributor.weeks, 'contributor')"
+            :options="getOptions('contributor')"
+          />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import {createApp} from 'vue';
+import {
+  Chart,
+  Title,
+  Tooltip,
+  Legend,
+  BarElement,
+  CategoryScale,
+  LinearScale,
+  TimeScale,
+  PointElement,
+  LineElement,
+  Filler,
+} from 'chart.js';
+import zoomPlugin from 'chartjs-plugin-zoom';
+import {Line} from 'vue-chartjs';
+import 'chartjs-adapter-dayjs-4/dist/chartjs-adapter-dayjs-4.esm';
+
+const {pageData} = window.config;
+
+Chart.register(
+  TimeScale,
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend,
+  PointElement,
+  LineElement,
+  Filler,
+  zoomPlugin
+);
+
+const sfc = {
+  components: {Line},
+  data: () => {
+    return {
+      isLoading: false,
+      totalStats: {},
+      repoLink: pageData.repoLink || [],
+      type: pageData.contributionType,
+      contributorsStats: [],
+    };
+  },
+  mounted() {
+    this.fetchGraphData();
+  },
+  computed: {
+    sortedContributors() {
+      return Object.values(this.contributorsStats)
+        .sort((a, b) =>
+          a.total_commits > b.total_commits ?
+            -1 :
+            a.total_commits === b.total_commits ?
+              0 :
+              1
+        )
+        .slice(0, 100);
+    },
+  },
+  methods: {
+    async fetchGraphData() {
+      this.isLoading = true;
+      fetch(`${this.repoLink}/contributors/data`)
+        .then((response) => response.json())
+        .then((data) => {
+          const {Total, ...rest} = data;
+          this.contributorsStats = rest;
+          this.totalStats = Total;
```

Comment:
```suggestion
          const {total, ...rest} = data;
          this.contributorsStats = rest;
          this.totalStats = total;
```

Reply:
`data` is not a list but an Object. It has a key named `Total` so it is necessary to access it with the same name. 
I should change the following line if we want this to be `total` but I don't think we really need to: https://github.com/go-gitea/gitea/blob/4ee972c59472c4db34b0a15485cbe432945a302f/models/contributors/contributors_graph.go#L56

Reply:
I think keys inside JSON data should be lowercase, so I suggest still changing it.